<img src="https://raw.githubusercontent.com/andre-marcos-perez/ebac-course-utils/main/media/logo/newebac_logo_black_half.png" alt="ebac-logo">

---

# **Módulo** | Análise de Dados: Fundamentos de Estatística
Caderno de **Exercícios**<br>
Professor [André Perez](https://www.linkedin.com/in/andremarcosperez/)

---

# **Tópicos**

<ol type="1">
  <li>Média e Variância;</li>
  <li>Ordem e Posição;</li>
  <li>Correlação.</li>
</ol>

---

# **Exercícios**

## 1\. Tráfego de São Paulo

Neste exercício, vamos continuar a analisar os dados de mobilidade urbana da cidade de São Paulo. A base de dados está neste [link](https://raw.githubusercontent.com/andre-marcos-perez/ebac-course-utils/develop/dataset/traffic.csv) e é uma copia do dado original, presente neste [link](https://archive.ics.uci.edu/ml/datasets/Behavior+of+the+urban+traffic+of+the+city+of+Sao+Paulo+in+Brazil). A base de dados contem a quantidade de acidentes ocorridos na cidade entre 14/12/09 e 18/12/09, das 07:00h ás 20:00h, agregados em intervalos de 30 minutos.

In [1]:
!wget -q "https://raw.githubusercontent.com/andre-marcos-perez/ebac-course-utils/develop/dataset/traffic.csv" -O traffic.csv

In [2]:
import pandas as pd

In [3]:
df = pd.read_csv('traffic.csv', sep=';')

In [4]:
df.head()

hour  bus_stoped  broken_truck  vehicle_excess  accident_victim  \
0     1           0             0               0                0   
1     2           0             0               0                0   
2     3           0             0               0                0   
3     4           0             0               0                0   
4     5           0             0               0                0   

   running_over  fire_vehicles  occurrence_freight  \
0             0              0                   0   
1             0              0                   0   
2             0              0                   0   
3             0              0                   0   
4             0              0                   0   

   incident__dangerous_freight  lack_electricity  fire  flooding  \
0                            0                 0     0         0   
1                            0                 0     0         0   
2                            0                 0     0         0   
3                            0                 0     0         0   
4                            0                 0     0         0   

   manifestations  defect_trolleybuses  tree_road  semaphore_off  \
0               0                    0          0              0   
1               0                    0          0              0   
2               0                    0          0              0   
3               0                    0          0              0   
4               0                    0          0              0   

   intermittent_emaphore slowness_traffic_%  
0                      0                4,1  
1                      0                6,6  
2                      0                8,7  
3                      0                9,2  
4                      0               11,1

- **Exemplo**: (para relembrar)

O código abaixo extrai a 13ª linha do arquivo que representa a 14ª meia hora do dia 14/12/09, contadas a partir das 07:00h, ou seja, todos os incidentes ocorridos na cidade no dia em questão entre as 13:30h e 14:00h. Vemos, por exemplo, que dois ônibus e dois caminhões ficaram paradados nas ruas da cidade, atrapalhando o tráfego.

In [5]:
df.iloc[[13]]

hour  bus_stoped  broken_truck  vehicle_excess  accident_victim  \
13    14           2             2               0                1   

    running_over  fire_vehicles  occurrence_freight  \
13             0              0                   0   

    incident__dangerous_freight  lack_electricity  fire  flooding  \
13                            0                 0     0         0   

    manifestations  defect_trolleybuses  tree_road  semaphore_off  \
13               0                    0          0              0   

    intermittent_emaphore slowness_traffic_%  
13                      0                4,9

### **1.1. Agregação**

Neste primeira atividade, você deve gerar um array NumPy por dia. Para cada array você deve somar todos os incidentes que aconteceram naquela meia hora. Sendo assim, cada array deve ter 27 posições, cada qual com a soma dos incidentes daquela meira hora.

**Dica**: Você deve remover a primeira e a última coluna.

**Dica**: Os arrays devem ter os seguintes valores:

```python
14: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 5, 2, 0, 0, 0, 0, 0, 1, 0, 0, 0, 2, 2, 3]
15: [3, 0, 0, 0, 2, 4, 2, 1, 3, 0, 2, 0, 2, 3, 2, 2, 4, 7, 6, 3, 0, 1, 1, 5, 4, 6, 0]
16: [1, 1, 2, 4, 4, 5, 3, 2, 4, 2, 1, 1, 3, 6, 2, 0, 2, 8, 4, 6, 4, 1, 2, 7, 15, 4, 0]
17: [4, 1, 1, 0, 0, 0, 2, 3, 2, 7, 2, 1, 7, 3, 6, 4, 3, 2, 2, 3, 2, 5, 2, 0, 1, 4, 15]
18: [10, 1, 1, 5, 5, 7, 3, 4, 3, 1, 2, 3, 2, 2, 6, 2, 2, 4, 2, 3, 0, 3, 3, 2, 3, 6, 0]
```

In [6]:
# resposta da questão 1.1

# remoção das colunas que não temos interresse
df.drop('hour', axis=1, inplace=True)
df.drop('slowness_traffic_%', axis=1, inplace=True)

In [7]:
import numpy as np

# arrays
dia_14 = np.array(df.iloc[0:27].T.values.reshape(16,27))
dia_15 = np.array(df.iloc[27:54].T.values.reshape(16,27))
dia_16 = np.array(df.iloc[54:81].T.values.reshape(16,27))
dia_17 = np.array(df.iloc[81:108].T.values.reshape(16,27))
dia_18 = np.array(df.iloc[108:137].T.values.reshape(16,27))

# dados de interesse
dia_14 = np.sum(dia_14, axis=0)
dia_15 = np.sum(dia_15, axis=0)
dia_16 = np.sum(dia_16, axis=0)
dia_17 = np.sum(dia_17, axis=0)
dia_18 = np.sum(dia_18, axis=0)

### **1.2. Métricas**

Para cada array você deve calcular as seguintes métricas:

 - **média**;
 - **desvio padrão**.

In [8]:
# resposta da questão 1.2
media_dia_14 = round(np.mean(dia_14), 2)
media_dia_15 = round(np.mean(dia_15), 2)
media_dia_16 = round(np.mean(dia_16), 2)
media_dia_17 = round(np.mean(dia_17), 2)
media_dia_18 = round(np.mean(dia_18), 2)

std_dia_14 = round(np.std(dia_14), 2)
std_dia_15 = round(np.std(dia_15), 2)
std_dia_16 = round(np.std(dia_16), 2)
std_dia_17 = round(np.std(dia_17), 2)
std_dia_18 = round(np.std(dia_18), 2)

print(f'Média dia 14: {media_dia_14}')
print(f'Média dia 15: {media_dia_15}')
print(f'Média dia 16: {media_dia_16}')
print(f'Média dia 17: {media_dia_17}')
print(f'Média dia 18: {media_dia_18}\n')

print(f'Desvio Padrão dia 14: {std_dia_14}')
print(f'Desvio Padrão dia 15: {std_dia_15}')
print(f'Desvio Padrão dia 16: {std_dia_16}')
print(f'Desvio Padrão dia 17: {std_dia_17}')
print(f'Desvio Padrão dia 18: {std_dia_18}')

Média dia 14: 0.59
Média dia 15: 2.33
Média dia 16: 3.48
Média dia 17: 3.04
Média dia 18: 3.15

Desvio Padrão dia 14: 1.19
Desvio Padrão dia 15: 2.02
Desvio Padrão dia 16: 3.05
Desvio Padrão dia 17: 3.05
Desvio Padrão dia 18: 2.19


### **1.3. Interpretação**

Baseado nos resultados da questão 1.2, responda:

> Qual dia apresenta a maior média de acidentes por meia hora?

**Resposta:** O "Dia 16" apresenta a maior média de acidentes por meia hora, com um valor de 3.48. Portanto, o "Dia 16" é o dia com a maior média de acidentes entre os dias fornecidos.

> Qual dia apresenta a menor variação de acidentes por meia hora?

**Resposta:** O menor desvio padrão é para o "Dia 14" com um valor de 1.19. Portanto, o "Dia 14" apresenta a menor variação de acidentes por meia hora entre os dias fornecidos.

---